Sources/HttpServer/Requests/RequestParser.swift

```diff
@@ -51,10 +52,21 @@ public class RequestParser {
         }
 
         let method = statusLineTokens[0]
-        let url = statusLineTokens[1]
+        let urlAndParams = statusLineTokens[1]
         let version = statusLineTokens[2]
+        let urlPlusParams = separateUrlFromParams(path: urlAndParams)
+        return (method: method, url: urlPlusParams.url, params: urlPlusParams.params, version: version)
+    }
 
-        return (method: method, url: url, version: version)
+    private func separateUrlFromParams(path: String) -> (url: String, params: [String]) {
+        guard path.contains("?") else {
```

Comment:
Out of curiosity, why did you use a guard clause here? Why not an if statement? It is not wrong or anything, just curious

Reply:
Just wanna try something different, that's all. I use if all the time and it took me a while to reason the conditional flow with "guard". Once I get the hang of it, I started using it everywhere :)